dhcp-protocol/src/main/java/org/anarres/dhcp/v6/messages/Dhcp6RelayMessage.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.anarres.dhcp.v6.messages;
+
+import com.google.common.base.Preconditions;
+import java.net.InetAddress;
+import javax.annotation.Nonnull;
+import org.anarres.dhcp.v6.options.Dhcp6Options;
+
+/**
+ * https://tools.ietf.org/html/rfc3315#section-7
+ *
+ *  @author marosmars
+ */
+public class Dhcp6RelayMessage extends Dhcp6Message {
+
+    private Dhcp6MessageType messageType;
+    private byte hopCount;
+    private InetAddress linkAddress;
+    private InetAddress peerAddress;
+
+    private Dhcp6Options options = new Dhcp6Options();
+
+    @Nonnull
+    public Dhcp6MessageType getMessageType() {
```

Comment:
This overrides a superclass method/field. If there is anything in the superclass which truly isn't shared, we should split into AbstractDhcp6Message (and interface Dhcp6Message?), rather than leaving fields in the supeclass which are semantically unusable. Otherwise, we should reuse the field.